source/extensions/filters/http/cache/http_cache.cc

```diff
@@ -68,29 +70,54 @@ LookupRequest::LookupRequest(const Http::RequestHeaderMap& request_headers, Syst
 size_t stableHashKey(const Key& key) { return MessageUtil::hash(key); }
 size_t localHashKey(const Key& key) { return stableHashKey(key); }
 
-// Returns true if response_headers is fresh.
-bool LookupRequest::isFresh(const Http::ResponseHeaderMap& response_headers) const {
-  if (!response_headers.Date()) {
-    return false;
+bool LookupRequest::requiresValidation(const Http::ResponseHeaderMap& response_headers) const {
+  // TODO: Store parsed response cache-control in cache instead of parsing it on every lookup
+  const ResponseCacheControl response_cache_control =
+      CacheHeadersUtils::responseCacheControl(response_headers.getCacheControlValue());
+
+  SystemTime::duration response_age =
+      timestamp_ - CacheHeadersUtils::httpTime(response_headers.Date());
+
+  if (response_cache_control.must_validate || request_cache_control_.must_validate ||
+      (request_cache_control_.max_age.has_value() &&
+       request_cache_control_.max_age.value() > response_age)) {
```

Comment:
It checks multiple (somehow) unrelated things. It checks if the response or request explicitly require revalidation or if the request's max-age condition requires revalidation. It could be extracted to 2 or 3 functions but then I think that defeats the purpose?

Reply:
Something like this can be clearer (perhaps the names could be better, but something like this):

`if (mustValidate(response_cache_control) || !maxAgeIsSatisfied(response_age))`

Reply:
Or perhaps just encapsulate the second bit:

`response_cache_control.must_validate || request_cache_control_.must_validate || !maxAgeIsSatisfied(response_age)`

Reply:
Maybe I can add similar ones for `min-fresh` and `max-stale` too, but with inverse logic (as that's what is always checked).
Something like
```
bool maxAgeViolated(response_age);
bool minFreshViolated(current_time, expiration_time);
bool maxStaleViolated(current_time, expiration_time);
```